pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -175,91 +170,10 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 		// using the helper.GetTargetNamespace function
 
 		// In Azure, the providerSecret is needed for Configure true or false
-		if hyd.Spec.Override == hypdeployment.InfraConfigureWithManifest {
+		if len(hyd.Spec.Override) == 0 || hyd.Spec.Override == hypdeployment.InfraConfigureWithManifest {
```

Comment:
So I think we need to update the description of the `Override` field in API?

Reply:
yep.

Reply:
Actually the point was to remove this. Right?

Reply:
yeah, I think we can remove the `Override`, and we need a new way to tell the controller to make the HostedCluster NodePool ORPHAN.
https://coreos.slack.com/archives/C02V40F8BH7/p1648822626187979?thread_ts=1648490337.830119&cid=C02V40F8BH7